python/packages/isce3/math/offsets_polyfit.py

```diff
@@ -0,0 +1,508 @@
+    from tie-point measurements by building a 2D polynomial design matrix up to
+    `degree`, solving with weighted normal equations, and iteratively removing
+    outliers using a w-test controlled by `crit_value`.
+
+    Parameters
+    ----------
+    data : (N, 6) array_like
+        Input tie-point array where each row is:
+        ``[id, line, pixel, dL, dP, corr_peak]``.
+        - *id* : int-like identifier.
+        - *line, pixel* : coordinates of the tie point.
+        - *dL, dP* : measured offsets in line and pixel directions.
+        - *corr_peak* : correlation peak or quality metric.
+    degree : int, optional
+        Total degree of the 2D polynomial (``1`` = affine, ``2`` = quadratic, ...).
+        Default is ``2``.
+    crit_value : float, optional
+        Critical value for the w-test used to flag outliers at each iteration.
+        Smaller values are stricter. Default is ``0.1``.
+    max_iterations : int, optional
+        Maximum number of outlier-rejection iterations. Default is ``50``.
+    minL, maxL : float, optional
+        Optional bounds on acceptable *line* offsets (units of `dL`).
+        Points with `dL` outside ``[minL, maxL]`` are removed before fitting.
+    minP, maxP : float, optional
+        Optional bounds on acceptable *pixel* offsets (units of `dP`).
+        Points with `dP` outside ``[minP, maxP]`` are removed before fitting.
+    prf, abw, rsr, rbw : float, optional
+        SAR system parameters (pulse repetition frequency, azimuth bandwidth,
+        range sampling rate, range bandwidth) used, if provided, to derive
+        prior standard deviations for weighting the line/pixel residuals.
+
+    Returns
+    -------
+    result : dict
+        Dictionary with the following keys:
+        - **coefL** : (M,) ndarray
+          Polynomial coefficients for the *line* direction, ordered consistently
+          with the design matrix construction, where
+          ``M = (degree + 1) * (degree + 2) // 2``.
+        - **coefP** : (M,) ndarray
+          Polynomial coefficients for the *pixel* direction.
+        - **inliers** : (K, 6) ndarray
+          Subset of `data` retained after robust outlier removal (``K <= N``).
+        - **removed_indices** : list of int
+          Row indices (w.r.t. the original `data`) that were removed as outliers
+          or by hard bounds.
+        - **degree** : int
+          Polynomial degree actually used.
+        - **design_nunk** : int
+          Number of unknown coefficients ``M``.
+    """
+
+    # Prior std for w-test
+    sigmaL = 0.15 / ((prf / abw) if ((prf is not None) and (abw is not None)) else 1.1)
+    sigmaP = 0.10 / ((rsr / rbw) if ((rsr is not None) and (rbw is not None)) else 1.1)
+
+    if None in (minL, maxL, minP, maxP):
+        minL, maxL = data[:, 1].min(), data[:, 1].max()
+        minP, maxP = data[:, 2].min(), data[:, 2].max()
+
+    Nunk, eps = ncoeffs(degree), 1e-8
+    removed_indices, iteration = [], 0
+
+    while True:
```

Comment:
```suggestion
    for iteration in itertools.count():
```
You can kill the need for hand-incrementing your loop counter and maintain an infinite loop by using `itertools.count()`

Reply:
good suggestion!